src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs

```diff
@@ -2418,6 +2480,46 @@ .maxstack  1
             Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "source()").WithArguments("System.MemoryExtensions", "AsSpan").WithLocation(2, 38));
     }
 
+    [Fact]
+    public void Conversion_Span_ReadOnlySpan_Retargeting()
+    {
+        var source1 = """
+            namespace System;
+            public readonly ref struct ReadOnlySpan<T>;
+            public readonly ref struct Span<T>
+            {
+                public static implicit operator ReadOnlySpan<T>(Span<T> span) => default;
+            }
+            """;
+        var comp1Ref = CreateCompilation(source1, targetFramework: TargetFramework.Mscorlib40)
+            .VerifyDiagnostics()
+            .ToMetadataReference();
+
+        var source2 = """
+            using System;
+            class C
+            {
+                ReadOnlySpan<int> M(Span<int> s) => s;
+            }
+            """;
+        var comp2 = CreateCompilation(source2, [comp1Ref]);
```

Comment:
When we're doing a retargeting test, it's usually good practice to explicitly specify the underlying TFM for the second compilation, so we can be sure that it's a different TFM from the assembly that will be retargeted.